pkg-config.cmake

```diff
@@ -478,23 +602,19 @@ ENDMACRO(PKG_CONFIG_APPEND_LIBS)
 #
 MACRO(PKG_CONFIG_USE_LCOMPILE_DEPENDENCY TARGET PREFIX)
 
-  # Make sure we do not override previous flags.
-  GET_TARGET_PROPERTY(CFLAGS ${TARGET} COMPILE_FLAGS)
-
-  # If there were no previous flags, get rid of the XYFLAGS-NOTFOUND
-  # in the variables.
-  IF(NOT CFLAGS)
-    SET(CFLAGS "")
-  ENDIF()
-
   # Transform semi-colon seperated list in to space separated list.
-  FOREACH(FLAG ${${PREFIX}_CFLAGS})
-    SET(CFLAGS "${CFLAGS} ${FLAG}")
-  ENDFOREACH()
-
-  # Update the flags.
-  SET_TARGET_PROPERTIES(${TARGET}
-    PROPERTIES COMPILE_FLAGS "${CFLAGS}")
+  IF(DEFINED ${PREFIX}_DEBUG_FOUND)
+    FOREACH(FLAG ${${PREFIX}_DEBUG_CFLAGS})
+      TARGET_COMPILE_OPTIONS(${TARGET} PUBLIC $<$<CONFIG:Debug>:${FLAG}>)
+    ENDFOREACH()
+    FOREACH(FLAG ${${PREFIX}_CFLAGS})
+      TARGET_COMPILE_OPTIONS(${TARGET} PUBLIC $<$<NOT:$<CONFIG:Debug>>:${FLAG}>)
+    ENDFOREACH()
+  ELSE()
+    FOREACH(FLAG ${${PREFIX}_CFLAGS})
+      TARGET_COMPILE_OPTIONS(${TARGET} PUBLIC ${FLAG})
```

Comment:
`TARGET_COMPILE_OPTIONS` should be replaced with `SET_TARGET_PROPERTIES` if it does work with generators, else this won't work on old CMake versions, e.g. the one on Ubuntu 12.04 (used on Travis).